duckduckgo/Android
# Support screen lock/unlock API

Task/Issue URL: https://app.asana.com/0/0/1206163779688838/f

### Description
Support [screen.lock](https://developer.mozilla.org/en-US/docs/Web/API/ScreenOrientation/lock) and [screen.unlock](https://developer.mozilla.org/en-US/docs/Web/API/ScreenOrientation/unlock) APIs

### Steps to test this PR
Before testing: apply a patch to C-S-S from https://github.com/duckduckgo/content-scope-scripts/pull/875

_Typical use case_
- [x] Open any video on YouTube and tap the full screen button - orientation should remain in portrait
- [x] Override PrivacyConfig to add an `"screenLock": "enabled"` to `webCompat` and ` "screenLock": {"state": "enabled"}` to `androidBrowserConfig`
- [x] Reload the YouTube page and press the full screen button again - orientation should switch to landscape
- [x] Use YouTube's button to exit full screen - orientation should switch back to portrait
  * Note: if you have auto-rotate ON, you just unlocked your phone, and it stayed flat on a desk - you may not have a default orientation set - in this case the orientation will remain in landscape upon full screen exit (this is the same behavior as in Chrome). If you orient your phone to portrait and try the exercise again, it should flip back automatically.
- [x] Test that disabling either the `webCompat` or the corresponding `androidBrowserConfig` disables the feature

  
_Web Share_
- [x] Smoke test web share as per https://github.com/duckduckgo/Android/pull/4027#discussion_r1433323915
  
 _Bonus_
 - [x]  Go to https://nshuba.github.io/test-site/screen.html and click Lock Orientation
 - [x] Text box should contain error indicating we need to be full screen before calling lock()
 - [x] Click Request Full Screen, select "landscape" from the dropbox and then click Lock Orientation
 - [x] Orientation should switch to landscape
 - [x] Test that the other orientation locks behave as expected (https://developer.mozilla.org/en-US/docs/Web/API/ScreenOrientation/lock)
 - [x] In the end, lock the orientation to any landscape or portrait-secondary and exit full screen (drag from the bottom and click "back")
 - [x] Orientation should change back to regular portrait
